### DISCUSSION TITLE
List with all-but-one greyed out

### DISCUSSION THREAD

**Discussion by collinmccarthy** [quarto-dev/quarto-cli#4572] [Q&A] [answered]
In a revealjs presentation, I have an outline slide like:

```
## Outline

- Part 1: Intro
- Part 2: Design
- Part 3: Performance
- Part 4: Next Steps
---
```

I would like to grey-out all but one item in the list to indicate what part of the presentation we're on. The closest I could get was:

```
## Outline

- Part 1: Intro

::: {.fragment .semi-fade-out}
- Part 2: Design
- Part 3: Performance
- Part 4: Next Steps
:::
---
```

But (a) it requires a mouse click/advance, and (b) it has an unnecessary newline (otherwise the fragment call just shows up as plain text). 

How should I be doing this? Apologies if the answer is the docs/discussions somewhere, I searched but couldn't find anything. Thank you!

**Comment by cscheid** [comment, marked as answer]
`.fragment` is meant to indicate a click for transitioning from one state to the next, and `.semi-fade-out` is meant to indicate "semi fade this out when clicked". We document this in our [Advanced Reveal](https://quarto.org/docs/presentations/revealjs/advanced.html#fragment-classes) page. So this is all working correctly, even if it's not what you expected. 

I think you're probably looking decorate the items themselves, so you can use spans and a custom class:

````
---
title: issue-4572
format: revealjs
---

## Outline

```{=html}
<style>
.semi-transparent {
  opacity: 0.5;
}
</style>
```

- Part 1: Intro
- [Part 2: Design]{.semi-transparent}
- [Part 3: Performance]{.semi-transparent}
- [Part 4: Next Steps]{.semi-transparent}
````

<img width="416" alt="image" src="https://user-images.githubusercontent.com/285675/221995061-9dfe8e4c-6738-47ae-a554-c764e74b1ac9.png">

That won't fade out the bullet items, unfortunately. AFAICT there's no simple way to wrap a single `li` element in markdown around a span so that the style matches the bullet pseudo-element. I suspect you could do fancier CSS things based on nth-element selectors, but you'll have to investigate and learn how to do that yourself.

**Reply by collinmccarthy** [reply]
This will work, thank you!

**Reply by collinmccarthy** [reply]
If I want to use this "semi-transparent" style across different slides, can I define that at the document level somehow?

**Reply by cscheid** [reply]
You can do that in the front matter. I'm linking from the html page, but this works on revealjs as well.

https://quarto.org/docs/output-formats/html-basics.html

**Reply by collinmccarthy** [reply]
Sorry I'm not seeing any examples of what the syntax would be in the front matter. I'm expecting something like 

```
format:
  revealjs:
    style:
      .semi-transparent:
         opacity: 0.5
```

But that's not working.

**Reply by collinmccarthy** [reply]
Obviously from the reference page those tags don't exist. Do I need to use both the default and my own CSS style sheet?

**Reply by cscheid** [reply]
```
title: "My document"
format:
  revealjs:
    css: choose-your-filename.css
```

**Reply by collinmccarthy** [reply]
Okay thanks. For anyone else, I created a .css file with 

```css
.half-transparent {
  opacity: 0.5;
}
.three-quarter-transparent {
  opacity: 0.25;
}
```

This worked, and the 3/4 transparency looked the best to me for this type of thing.

**Comment by mcanouil** [comment]
May I suggest the following Quarto extension which sole purpose is to do what you want to do.

https://github.com/andrie/reveal-auto-agenda

**Reply by collinmccarthy** [reply]
This looks great, thanks! If my workflow gets more complicated I'll definitely use this.
